app/deployments/routes.py

```diff
@@ -1420,17 +1500,141 @@ def createdep():
     else:
         remove_sla_from_template(template)
 
+    create_dep_method(
+        source_template,
+        selected_template,
+        additionaldescription,
+        inputs,
+        form_data,
+        template,
+        template_text,
+    )
+
+    return redirect(url_for(SHOW_DEPLOYMENTS_ROUTE))
+
+
+@deployments_bp.route("/<depid>/retry")
+@auth.authorized_with_valid_token
+def retrydep(depid=None):
+    """
+    A function to retry a failed deployment.
+    Parameters:
+    - depid: str, the ID of the deployment
+    """
+    tosca_info, _, _ = tosca.get()
+    access_token = iam.token["access_token"]
```

Comment:
What if we don't have the "access_token" key? Should we use `try/catch` of some guard clause to raise an exception if we don't have the token?